primitives/src/pcs/mod.rs

```diff
@@ -58,36 +62,36 @@ pub trait PolynomialCommitmentScheme<E: PairingEngine> {
     /// Input both `supported_degree` for univariate and
     /// `supported_num_vars` for multilinear.
     fn trim(
-        srs: &Self::SRS,
+        srs: impl Borrow<Self::SRS>,
         supported_degree: usize,
         supported_num_vars: Option<usize>,
     ) -> Result<(Self::ProverParam, Self::VerifierParam), PCSError>;
 
     /// Generate a commitment for a polynomial
     fn commit(
-        prover_param: &Self::ProverParam,
```

Comment:
maybe fore the first occurrence of `srs` and `prover_param`, let's add some note in the code comments that: 

```
\\\ ## Note on function signature 
\\\ Usually, data structure like SRS and ProverParam are huge and users might wish to keep them in heap using different kinds of smart pointers (instead of only in stack)
\\\ therefore our `impl Borrow<_>` interface allows for passing in any pointer type, e.g.:
\\\ `trim(srs: &Self::SRS, ..)` or `trim(srs: Box<Self::SRS>, ..)` or `trim(srs: Arc<Self::SRS>, ..)` etc.
```